helix-term/src/commands/lsp.rs

```diff
@@ -325,6 +325,16 @@ pub fn symbol_picker(cx: &mut Context) {
     let doc = doc!(cx.editor);
 
     let language_server = language_server!(cx.editor, doc);
+
+    if matches!(
+        language_server.capabilities().workspace_symbol_provider,
+        None | Some(lsp::OneOf::Left(false))
+    ) {
+        cx.editor
+            .set_status("Document symbols are not enabled by the Language Server");
+        return;
+    }
```

Comment:
I'd rather match on the success case, than negative since it's easy to miss a case. We should also decide if we want to handle this inside commands, or inside the LSP client: 

https://github.com/helix-editor/helix/blob/83f177d2701b0ec58abcb47c2a1ee133e72342e4/helix-lsp/src/client.rs#L760-L766

 I was thinking these calls could return a `Result<Future, Error>` so `Err::Unavailable`/`Unsupported` could be a generic way of reporting this back to the caller